src/lava/magma/core/model/py/model.py

```diff
@@ -30,7 +30,7 @@ class AbstractPyProcessModel(AbstractProcessModel, ABC):
         du: int =          LavaPyType(int, np.uint16, precision=12)
     """
 
-    def __init__(self):
+    def __init__(self, proc_params: ty.Dict[str, ty.Any] = {}):
```

Comment:
See other comment in Builder.
There is probably not even a need to initialize with an empty dict. This can be a mandatory argument.

If a ProcModel does not have an __init__ method then the AbstractProcModel or AbstractPyProcModel __init__ will be called with possibly an empty proc_params of the corresponding process.

If a ProcModel has an __init__ method then it needs to deal with proc_params by passing it to the parent __init___ method.

Reply:
Removed default initialization